src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -542,27 +544,27 @@ public void RecordException(Exception ex)
         /// <summary>
         /// Set the test result based on the type of exception thrown
         /// </summary>
-        /// <param name="ex">The exception that was thrown</param>
+        /// <param name="exception">The exception that was thrown</param>
         /// <param name="site">The FailureSite to use in the result</param>
-        public void RecordException(Exception ex, FailureSite site)
+        public void RecordException(Exception exception, FailureSite site)
         {
-            if (ex is NUnitException)
-                ex = ex.InnerException;
+            if (exception is NUnitException)
```

Comment:
I think we should add `&& exception.InnerException != null` here too, and throw ArgumentNullException at the top of the method for `exception`.

@nunit/framework-team Does anyone know why we aren't unwrapping `TargetInvocationException` too like we do in the other `RecordException` overload?

Reply:
I'm pretty sure I added the TargetInvocationException line - and most likely missed that the code was duplicated. I wonder if I can de-duplicate?

@CharliePoole will be happy with this, as we've written the duplicate code first, and are removing it second! 😄 😉

Reply:
De-duping sounds good!

Reply:
Is there any reason why `RecordException(Exception)` does not directly invoke `RecordException(Exception, FailureSite)` or the other way around? I can tackle that in this PR if you can tell me why much of the functionality inside the methods are duplicated in the first place. 😃